test/Splunk.OpenTelemetry.AutoInstrumentation.IntegrationTests/SelfContainedTests.cs

```diff
@@ -0,0 +1,128 @@
+// limitations under the License.
+// </copyright>
+
+using FluentAssertions;
+using Splunk.OpenTelemetry.AutoInstrumentation.IntegrationTests.Helpers;
+using Xunit.Abstractions;
+
+namespace Splunk.OpenTelemetry.AutoInstrumentation.IntegrationTests;
+
+[Trait("Category", "NuGetPackage")]
+public sealed class SelfContainedTests : TestHelper
+{
+    private readonly string _selfContainedAppDir;
+
+    public SelfContainedTests(ITestOutputHelper output)
+        : base("SelfContained", output, "nuget-package")
+    {
+        var nonSelfContainedOutputDir = EnvironmentHelper
+            .GetTestApplicationApplicationOutputDirectory();
+
+        // The self-contained app is going to have an extra folder before it: the one
+        // with a RID like "win-x64", "linux-x64", etc.
+        var childrenDirs = Directory.GetDirectories(nonSelfContainedOutputDir);
+        childrenDirs.Should().ContainSingle();
+
+        _selfContainedAppDir = childrenDirs[0];
+    }
+
+    [Fact]
+    public void InstrumentExecutable()
+    {
+        RunAndAssertHttpSpans(() =>
+        {
+            var instrumentationTarget = Path.Combine(_selfContainedAppDir, EnvironmentHelper.FullTestApplicationName);
+            instrumentationTarget = EnvironmentTools.IsWindows() ? instrumentationTarget + ".exe" : instrumentationTarget;
+            RunInstrumentationTarget(instrumentationTarget);
+        });
+    }
+
+#if !NETFRAMEWORK
+    [Fact]
+    public void InstrumentDll()
+    {
+        RunAndAssertHttpSpans(() =>
+        {
+            var dllName = EnvironmentHelper.FullTestApplicationName.EndsWith(".exe")
+                ? EnvironmentHelper.FullTestApplicationName.Replace(".exe", ".dll")
+                : EnvironmentHelper.FullTestApplicationName + ".dll";
+            var dllPath = Path.Combine(_selfContainedAppDir, dllName);
+            RunInstrumentationTarget($"dotnet {dllPath}");
+        });
+    }
+#endif
+
+    private void RunInstrumentationTarget(string instrumentationTarget)
+    {
+        var instrumentationScriptExtension = EnvironmentTools.IsWindows() ? ".cmd" : ".sh";
+        var instrumentationScriptPath =
+            Path.Combine(_selfContainedAppDir, "splunk-launch") + instrumentationScriptExtension;
+
+        Output.WriteLine($"Running: {instrumentationScriptPath} {instrumentationTarget}");
+
+        using var process = InstrumentedProcessHelper.Start(instrumentationScriptPath, instrumentationTarget, EnvironmentHelper);
+        using var helper = new ProcessHelper(process);
+
+        process.Should().NotBeNull();
+
+        bool processTimeout = !process!.WaitForExit((int)Helpers.Timeout.ProcessExit.TotalMilliseconds);
+        if (processTimeout)
+        {
+            process.Kill();
+        }
+
+        Output.WriteLine("ProcessId: " + process.Id);
+        Output.WriteLine("Exit Code: " + process.ExitCode);
+        Output.WriteResult(helper);
+
+        processTimeout.Should().BeFalse("Test application timed out");
```

Comment:
nit: I think `because` part is a bit misleading - it should be something like "test application should not timeout".
Same goes for the assertion below.

Reply:
Thanks for calling my attention to that, I wasn't aware of the `because` message on fluent assertions.

Now, it will be:
```
Expected (processTimeout) to be false because test application should NOT have timed out, but found True.
```

8357c3b198e047673d4dd9758c78a54572486efe